File: suraj_24b0907/week7/houserobber.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    int rob(vector<int>& nums) {
        int n=nums.size();
        if(n==1) return nums[0];
        vector<int> v={nums[0],max(nums[0],nums[1])};
        for(int i=2;i<n;i++){
            v.push_back(max(v[i-2]+nums[i],v[i-1]));
        }
        return v[n-1];
    }
};

File: Chavda_23B0368/Week_7/House_robber.java
package Week_7;

class Solution {
    public int rob(int[] nums) {
        if(nums.length == 0) return 0;
        if(nums.length == 1) return nums[0];
        int n = nums.length;
        int prev2 = nums[0];
        int prev = Math.max(prev2, nums[1]);
        int curr = 0;
        for(int i = 2; i < n; i++) {
            curr = Math.max(prev, prev2+nums[i]);
            prev2 = prev;
            prev = curr;
        }

        return prev;
    }
}